sdk/core/core-client/src/serializer.ts

```diff
@@ -518,8 +552,34 @@ function serializeDictionaryType(
   }
   const tempDictionary: { [key: string]: any } = {};
   for (const key of Object.keys(object)) {
-    tempDictionary[key] = serializer.serialize(valueType, object[key], objectName + "." + key);
+    const serializedValue = serializer.serialize(valueType, object[key], objectName);
+    // If the element needs an XML namespace we need to add it within the $ property
+    if (isXml && valueType.xmlNamespace) {
+      const xmlnsKey = valueType.xmlNamespacePrefix
+        ? `xmlns:${valueType.xmlNamespacePrefix}`
+        : "xmlns";
+      // If the value is an object the object's properties need to be siblings of the $ property
+      if (valueType.type.name === "Composite") {
+        tempDictionary[key] = { ...serializedValue, $: { [xmlnsKey]: valueType.xmlNamespace } };
+        continue;
+      } else {
+        // When the value is not an object, it has to go under _
+        tempDictionary[key] = { _: serializedValue, $: { [xmlnsKey]: valueType.xmlNamespace } };
+        continue;
+      }
+    }
+
+    // Just add the value when we are not serializing XML or it doesn't need a namespace
```

Comment:
nitpick: maybe remove this comment since the value is added in both cases?

Reply:
This should only be assigned when valueType.type.name is not Composite. I'm moving this inside an else instead of relying on the continues to be safe and improve readability

Reply:
ah missed continue...

Reply:
now the code here is very similar to `getXmlObjectValue()`, any possibility of refactoring to call `getXmlObjectValue()`?

Reply:
Oh yeah! you're right :smile: thanks for suggesting it!